### PR TITLE
Fix Duplicate Satellite Join Token Request

### PR DESCRIPTION
**Pull request by bupd**
- Fixes: #284 

## Description
Moves satellite existence check before SPIRE operations in /api/join-tokens handler. Returns 409 Conflict if satellite already exists, preventing unnecessary SPIRE token/workload creation.

## Additional context
none

---
## Summary by cubic
Reject duplicate satellite join-token requests with a 409 Conflict before any SPIRE operations, and wrap satellite creation in a DB transaction to avoid race conditions.

- **Bug Fixes**
  - Check satellite existence first; wrap DB ops in a transaction; return 409 if the name already exists (including race), 500 on DB errors.
  - Create satellite record, robot account, and default config only for new satellites.
  - Add handler tests for missing satellite name, empty/invalid selectors, missing SPIRE client (503), and duplicate satellite (409).

- **Dependencies**
  - Add github.com/DATA-DOG/go-sqlmock for DB mocking in tests.

<sup>Written for commit c4ddf70f1b30cd55d671ec1bdbce12cf2f37c38f. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved join-token creation: ensures satellite uniqueness, uses a transactional flow to avoid partial provisioning, and returns appropriate error codes for conflicts and failures.
* **Tests**
  * Added unit tests covering validation errors, missing services, and duplicate-satellite scenarios to prevent regressions.
* **Chores**
  * Updated build dependencies.

### CODE REVIEW

File: ground-control/internal/server/spire_handlers.go [hunk 1]
```diff
@@ -150,40 +172,66 @@ func (s *Server) createJoinTokenHandler(w http.ResponseWriter, r *http.Request)
...
+		return
+	}
+
+	if err := tx.Commit(); err != nil {
+		log.Printf("Join token: Failed to commit transaction for %s: %v", req.SatelliteName, err)
+		HandleAppError(w, &AppError{
+			Message: "Failed to commit satellite creation",
+			Code:    http.StatusInternalServerError,
+		})
+		return
```
[the remark this answers is not in the document]
Comment (bupd) [reply ground-control/internal/server/spire_handlers.go:234]: lets track this in a separate issue.

[the remark this answers is not in the document]
Comment (bupd) [reply ground-control/internal/server/spire_handlers.go:234]: @coderabbitai - create an issue for this

[the remark this answers is not in the document]
Comment (bupd) [reply ground-control/internal/server/spire_handlers.go:234]: yes create an issue

File: ground-control/internal/server/spire_handlers.go [hunk 2]
```diff
@@ -150,40 +172,66 @@ func (s *Server) createJoinTokenHandler(w http.ResponseWriter, r *http.Request)
...
+		HandleAppError(w, &AppError{
+			Message: "Failed to create satellite record",
+			Code:    http.StatusInternalServerError,
+		})
+		return
+	}
+	log.Printf("Join token: Created satellite record for %s", req.SatelliteName)
+
+	// Create robot account and link default config for new satellites
+	if _, _, robotErr := ensureSatelliteRobotAccount(r, txQueries, satellite); robotErr != nil {
```
[the remark this answers is not in the document]
Comment (bupd) [reply ground-control/internal/server/spire_handlers.go:209]: create an issue for this - since this falls in the atomic transaction criteria
